### DISCUSSION TITLE
GUI for lncrawl

### DISCUSSION THREAD

**Discussion by Gamedemons** [lncrawl/lightnovel-crawler#1419] [Show and tell]
I made a GUI version named [Darknovel Crawler](https://github.com/Gamedemons/Darknovel-Crawler) using lncrawler. I also made a pre release, although it is not fully complete its downloads option is almost implemented. Can you check it out [here](https://github.com/Gamedemons/Darknovel-Crawler/releases)

**Comment by Gamedemons** [comment]
Also if you can give me any tips it would be great

**Comment by dipu-bd** [comment]
I was checking the code. I could not run because of some compile issues. 

You are using the horizontal bar for type declaration. like: `NovelFromSource | None` in many of the files. It is not allowed in python3. You have to refactor the code like this:

- Instead of `TypeName | None`, you can use `Optional[TypeName]`
- `Type1 | Type2 | Type3` can be written as `Union[Type1, Type2, Type3]`

**Reply by jere344** [reply]
I think you replied to the wrong person.
I forgot about pipes, it's a python 3.10 feature, I will replace them.

**Reply by dipu-bd** [reply]
I kept open two tabs. Forgot to check which tab I was in before writing the reply. 🐸

**Comment by dipu-bd** [comment]
My suggestion for the frontend is that instead of using raw html and js files, use a js framework like vue or react. You can use the flask server as an API backend and serve the static assets we get from bulding a vue and react app as the frontend. It will be a lot more maintainable in that case.

**Reply by jere344** [reply]
I will check how can I do that

**Reply by jere344** [reply]
So I went to see how react works. I think I can make a react application that uses flask in the backend (like this guy https://www.youtube.com/watch?v=7LNl2JlZKHA).
But I'm not sure I understand about serving statics assets. Do you mean pre generate html files and serve them with flask? It seems complicated to me since you would need one file per chapter.

Also could you give me the log about the compile issue ? I tried on linux and windows and had no issues but perhaps I forgot some dependencies that I already had installed.

**Comment by Gamedemons** [comment]
I too think you replied to a wrong person. Im @Gamedemons. The UI @jere344 made is different from the one I made.[Here](https://github.com/Gamedemons/Darknovel-Crawler/releases ) is the one I made

**Reply by dipu-bd** [reply]
Oh, you are right. Sorry about that.

**Reply by Gamedemons** [reply]
> Oh, you are right. Sorry about that.

any thoughts now ??
